### ISSUE TITLE
ColumnSelectingTransformer exported onnx model doesn't drop input columns coming from input dataview

### ISSUE THREAD

**Issue by antoniovs1029** [dotnet/machinelearning#4970]
The `ColumnSelectingTransformer` has the ability to drop columns that aren't desired by the user. The Onnx exported model eliminates those columns from the Onnx model graph output, but this isn't enough to remove them from the output `Schema`, so even if they're removed from the onnx model, those columns are still there in the output after using the `ApplyOnnxModel` method. The reason for this is that the `OnnxTransformer` (which is used by the `ApplyOnnxModel` method) is a `RowToRowTransformer`, and such transformers don't have the capacity to drop columns, only to add them to the input DataView.

### Code
The existing test for ColumnSelectingTransformer can be used to notice this issue
https://github.com/dotnet/machinelearning/blob/22c7ac8921fa0846717662181b334de3b41e9932/test/Microsoft.ML.Tests/OnnxConversionTest.cs#L1707-L1765

By setting a breakpoint, and inspecting into the schema of the outputs, I get the following result:
![image](https://user-images.githubusercontent.com/38739674/77539211-385fb900-6e5e-11ea-9bd2-a276697bf264.png)

It shows that the undesired columns were correctly dropped by the `ColumnSelectingTransformer`, but not by the `OnnxModel`. This test should also include comparing both schemas, to make sure that the onnxmodel is actually dropping the undesired columns.

**Comment by antoniovs1029** [comment]
Notice that this is a problem in ML.NET and not in the Onnx model exported from `ColumnSelectingTransformer`, since the model itself doesn't include the undesired output columns as outputs. This is relevant, because if a user were to use the exported onnx model outside of ML.NET (e.g., inside Python, by using `OnnxRuntime`), then the output of the model will actually "drop" the undesired inputs.

I've tested this with NimbusML ORT test for `ColumnSelector`, and it works correctly in that case. But it doesn't work correctly when using `OnnxRunner`, because of the same reasons it doesn't work on ML.NET.

**Comment by antoniovs1029** [comment]
Notice that there exists a simple workaround where the user could actually add a ColumnSelectingTransformer after applying the Onnx model, to remove whatever columns the onnx model didn't remove.

Notice that this workaround can also work in NimbusML, by adding a ColumnSelector after an OnnxRunner.

**Comment by antoniovs1029** [comment]
After discussing this offline with @harishsk  and @yaeldekel we've come to the conclusion that it isn't clear if this issue should be fixed or not (i.e. if it's a real issue or not). So we've leaned towards not fixing it, unless it becomes a problem for real users.

The work that I made towards fixing it on PR #5012 has been saved in a new branch on this repository:
https://github.com/dotnet/machinelearning/tree/onnx-columnselector

In case we want to revisit this issue, it might be useful to take the code in that branch and the comments on the PR.
